### DISCUSSION TITLE
Manually trigger embedding for the current file

### DISCUSSION THREAD

**Discussion by zeroliu** [brianpetro/obsidian-smart-connections#235] [Ideas]
Hi @brianpetro ,

Can you please share the high-level behavior of embedding for new notes? When will they trigger if the content is updated? The plugin doesn't rerun the embedding API frequently, probably because of saving the cost. When I start a new note, I am often interested to see other notes about similar topics. However, the smart view won't provide meaningful connections if the embedding is created when the note is empty. I'd love to see a button in the side panel to manually trigger the embedding for the current note and refresh the smart view result. This feature can make the plugin more friendly for creating new notes. 

Please let me know your thoughts!

Thanks

**Comment by brianpetro** [comment]
You can run the "Find Smart Connections" command when a note is open to trigger re-embedding the note. 

There is also a "render log" setting you can toggle on that will print an overview of which notes were embedded and which have changed but not enough to pass the threshold for re-embedding.

🌴

**Reply by zeroliu** [reply]
Thanks! Didn't know this feature existed. Surfacing it to the visual UI might still be helpful for people who don't check command.

There might be a bug in the threshold checking. For example, no embedding is run when I deleted all the content in a note. Same when I added the content back to the note.

**Reply by brianpetro** [reply]
If you run the command after making the change it should process the re-embedding. Similarly, closing and then re-opening the file should work.
